pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandler.java

```diff
@@ -359,7 +359,8 @@ private boolean shouldDisableDictionary(String column, ColumnMetadata existingCo
     }
 
     // Sorted columns should always have a dictionary.
-    if (existingColumnMetadata.isSorted()) {
+    if (existingColumnMetadata.isSorted()
+        && !(existingColumnMetadata.getCardinality() == 1 && existingColumnMetadata.isAutoGenerated())) {
```

Comment:
columns which are auto-generated as part of segment reload after adding a new column always have `cardinality: 1` because the default null value for the column is the sole value in the segment.

When columns have cardinality of 1, they are inferred to be sorted.  It's possible a better fix exists where the noDictColumns can be referenced at the time that "isSorted" is determined.